renet_steam_transport/src/transport/client.rs

```diff
@@ -0,0 +1,132 @@
+use std::{sync::Arc, time::Duration};
+
+use super::{Transport, MAX_MESSAGE_BATCH_SIZE};
+use bevy_utils::syncunsafecell::SyncUnsafeCell;
+use renet::RenetClient;
+use steamworks::{
+    networking_sockets::{InvalidHandle, NetConnection},
+    networking_types::{NetConnectionEnd, NetConnectionStatusChanged, NetworkingConnectionState, NetworkingIdentity, SendFlags},
+    CallbackHandle, ClientManager, SteamId,
+};
+#[cfg_attr(feature = "bevy", derive(bevy_ecs::system::Resource))]
+pub struct SteamClientTransport {
+    connection: NetConnection<ClientManager>,
+    status: NetworkingConnectionState,
+    callback_handle: Option<CallbackHandle<ClientManager>>,
+    disconnect_reason: NetConnectionEnd,
+    pending: Arc<SyncUnsafeCell<Vec<NetConnectionStatusChanged>>>,
+}
```

Comment:
```suggestion
pub struct SteamClientTransport {
    connection: NetConnection<ClientManager>,
    status: NetworkingConnectionState,
    callback_handle: CallbackHandle<ClientManager>,
    disconnect_reason: Option<NetConnectionEnd>,
    status_change_receiver: Receiver<NetConnectionStatusChanged>,
}
```

We can save the callback inside the `new`, instead of having to call `register_callback`. We can use sync::channel for the callback communication instead of Arc<SyncUnsafeCell<Vec<NetConnectionStatusChanged>>>.

```rust
 pub fn new(..) -> Result<Self, InvalidHandle> {
    let (sender, status_change_receiver) = mpsc::channel();
    let callback_handle = steam_client.register_callback::<NetConnectionStatusChanged, _>(move |event| {
         sender.send(event)
     });
 }
 ```
 
 Then inside the `update` we can call:
 `self.status_change_receiver.try_receive`

Reply:
Wow that is so much cleaner, thank you! :) But there is actually one problem with the Bevy Ressource, because the Reciever does not implement the trait `Sync`

Reply:
Okay i implemented it with the help of tokio, but i now need to safe the  `update` and `send_packets` functions with a guard function if the connection is connected.. im not sure how good that is.

Reply:
hmm maybe we could only guard the log error case, that would safe us a lot of checks, what are your thoughts on it?